tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -243,30 +216,152 @@ private static void AddAppsecSpecificInstrumentations()
 
             try
             {
-                Log.Debug("Sending CallTarget appsec derived integration definitions to native library.");
+                Log.Debug("Adding CallTarget appsec derived integration definitions to native library.");
                 var payload = InstrumentationDefinitions.GetDerivedDefinitions(InstrumentationCategory.AppSec);
                 NativeMethods.InitializeProfiler(payload.DefinitionsId, payload.Definitions);
-                foreach (var def in payload.Definitions)
-                {
-                    def.Dispose();
-                }
+                derived = payload.Definitions.Length;
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, ex.Message);
+            }
+
+            Log.Information($"{defs} AppSec definitions and {derived} derived definitions added to the profiler.");
+        }
 
-                Log.Information<int>("The profiler has been initialized with {count} AppSec derived definitions.", payload.Definitions.Length);
+        private static void RemoveAppsecSpecificInstrumentations()
+        {
+            int defs = 0, derived = 0;
+            try
+            {
+                Log.Debug("Adding CallTarget AppSec integration definitions to native library.");
+                var payload = InstrumentationDefinitions.GetAllDefinitions(InstrumentationCategory.AppSec);
+                NativeMethods.RemoveCallTargetDefinitions(payload.DefinitionsId, payload.Definitions);
+                defs = payload.Definitions.Length;
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, ex.Message);
+            }
+
+            try
+            {
+                Log.Debug("Adding CallTarget appsec derived integration definitions to native library.");
```

Comment:
```suggestion
                Log.Debug("Removing CallTarget appsec derived integration definitions from native library.");
```

Reply:
Fixed in 270b271d3de3380c939aaf7e1881a9d7bbf41fd6